### PR TITLE
Cleanup retry list

### PR DESCRIPTION
**Pull request by HaoK**
Remove 2 tests from retry now that they have been fixed by https://github.com/dotnet/aspnetcore/issues/42982

### CODE REVIEW

**Comment by HaoK** [comment]
Will take this PR to clean up and remove all tests that haven't failed/needed a retry in the last 45 days

**Comment by HaoK** [comment]
AttributeRouting_RouteNameTokenReplace_Reachable 809 passes in 45 days 100%

**Comment by HaoK** [comment]
So also removing from retry:

AttributeRouting_RouteNameTokenReplace_Reachable 
ServerCertificateSelector_Invoked 
VariableMultipleStreamsInSequence 

As these 3 have 100% pass rates over the last 45 days

File: eng/test-configuration.json [hunk 1]
```diff
@@ -5,22 +5,16 @@
...
     {"testName": {"contains": "TraceIdentifierIsUnique"}},
     {"testName": {"contains": "LocationChangingHandlers_CannotCancelTheNavigationAsynchronously_UntilReturning"}},
     {"testName": {"contains": "ClientAbortingConnectionImmediatelyIsNotLoggedHigherThanDebug"}},
-    {"testName": {"contains": "Listen_Http3AndSocketsCoexistOnSameEndpoint_AltSvcEnabled_Upgrade"}},
-    {"testName": {"contains": "Listen_Http3AndSocketsCoexistOnSameEndpoint_AltSvcDisabled_NoUpgrade"}},
     {"testName": {"contains": "WebConfigAppendsHostingStartup_OutOfProcess"}},
     {"testName": {"contains": "HttpsConnectionClosedWhenResponseDoesNotSatisfyMinimumDataRate"}},
     {"testName": {"contains": "BidirectionalStream_ServerReadsDataAndCompletes_GracefullyClosed"}},
     {"testName": {"contains": "POST_ServerAbort_ClientReceivesAbort"}},
-    {"testName": {"contains": "VariableMultipleStreamsInSequence"}},
```
Comment (dougbu) [inline eng/test-configuration.json:23]: Just wondering if all of these tests are actually going to stop retrying❔ For example, are some covered by the blanket `Http3RequestTests`, `Http3TlsTests`, or `IIS` rules❔

**Review by dougbu — APPROVED** [review]

File: eng/test-configuration.json (hunk 1 above)
Comment (HaoK) [reply eng/test-configuration.json:23]: That's somewhat orthogonal, I'm just trying to reduce the number of lines on this list right now.  Keeping this file manageable is the main intent.

Comment (HaoK) [reply eng/test-configuration.json:23]: Also retries are harmless if the test doesn't fail very often anyways, but this list growing huge is a maintenance headache no matter what

**Pull request merged**